horizomer/misc/taxonomy/tree_by_taxdump.py

```diff
@@ -0,0 +1,142 @@
+# this script generates a tree of a set genomes based on their NCBI taxonomy
+# (taxdump) annotations
+# usage: python me.py nodes.dmp genome2taxid.tsv output.nwk
+
+import sys
+from skbio import TreeNode
+
+# read NCBI taxonomy hierarchies (nodes.dmp)
+taxdump = {}
+with open(sys.argv[1], 'r') as f:
+    for line in f:
+        # format of nodes.dmp:
+        # taxid <tab> | <tab> parent taxid <tab> | <tab> more info...
+        l = line.rstrip('\r\n').split('\t|\t')
+        taxdump[l[0]] = {'parent': l[1], 'children': set()}
+
+# identify root of the tree and children of all nodes
+root = None
+for tid in taxdump:
+    pid = taxdump[tid]['parent']
+    # at root, the taxid and its parental taxid are both 1
+    if tid == pid:
+        root = tid
+    elif pid in taxdump:
+        taxdump[pid]['children'].add(tid)
+    else:
+        raise ValueError('TaxID %s is not defined.' % pid)
+
+# create the tree from root
+tree = TreeNode(root)
+
+
+# iteratively attach child nodes to parent node
+def iter_node(node):
+    for cid in taxdump[node.name]['children']:
+        child = TreeNode(cid)
+        node.extend([child])
+        iter_node(child)
```

Comment:
Can we modularize this, and make taxdump a parameter argument?  Once that is done, can we create a unittest for this?

Reply:
@mortonjt Your suggestions are being full-filled. Please check out my new code.